### PR TITLE
[Feat][kubectl-plugin] Add dynamic shell completion for kubectl ray session

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

Implement dynamic shell completion for `kubectl ray session` command.

Note: I haven't found a proper way to mock the cmd Factory to get custom resources for testing.

Screencast:

https://github.com/user-attachments/assets/b1df97c1-2be2-4985-a2f0-ec82b51db98a

## Related issue number

N/A

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by MortalHappiness** [comment]
@kevin85421 @andrewsykim @chiayi PTAL

**Comment by kevin85421** [comment]
The demo video looks good! Is it only for `session`, or do other commands like `get` also support it?

**Comment by MortalHappiness** [comment]
@kevin85421 Currently only session command.

**Comment by andrewsykim** [comment]
@MortalHappiness the autocomplete looks awesome, can't wait to try it! 

>Currently only session command. 

@chiayi can we look into implementing similar autocompletion for `kubectl ray logs`?

File: kubectl-plugin/pkg/util/completion/completion.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	}
+}
+
+// RayClusterCompletionFunc Returns a completion function that completes RayCluster resource names.
+func RayClusterCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return completion.ResourceNameCompletionFunc(f, string(util.RayCluster))
+}
+
+// RayJobCompletionFunc Returns a completion function that completes RayJob resource names.
+func RayJobCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
```
Comment (andrewsykim) [inline kubectl-plugin/pkg/util/completion/completion.go:37]: Is `RayJobCompletionFunc` and `RayServiceCompletionFunc` still needed? It looks like `RayClusterResourceNameCompletionFunc` handles resource type already

Comment (MortalHappiness) [reply kubectl-plugin/pkg/util/completion/completion.go:37]: These 3 functions `RayClusterCompletionFunc`, `RayJobCompletionFunc`, and `RayServiceCompletionFunc` are not used in this PR. But I think they may be useful for implementing autocompletion for other commands. Let me introduce the differences:

`RayClusterResourceNameCompletionFunc`:
- If `toComplete` does not have slash, complete it from either RayCluster names or resource types (`raycluster/`, `rayjob/`, `rayservice/`)
- If `toComplete` starts with resource types and slash, complete it with the corresponding resource name (e.g. `rayjob/sample-rayjob`)
- Examples:
  - `r<Tab>` -> `raycluster-1`, `raycluster-2`, `raycluster/`, `rayjob/`, `rayservice/`
  - `m<Tab>` -> `my-raycluster-1`, `my-raycluster-2`
  - `rayjob/<Tab>` -> `rayjob/rayjob-1`, `rayjob/rayjob-2`

`RayClusterCompletionFunc`, `RayJobCompletionFunc`, `RayServiceCompletionFunc`:
- These functions only complete the resource name of the corresponding resource. For example, for `RayJobCompletionFunc`, the completion result will not include `rayjob/`, only the name of RayJobs are presented to the users.
- So these function can be used by other commands. For example, `kubectl ray logs` can use `RayClusterCompletionFunc`. cc @chiayi 
- Examples (`RayClusterCompletionFunc`):
  - `r<Tab>` -> `raycluster-1`, `raycluster-2`

Note that the `r<Tab>` results are different for `RayClusterResourceNameCompletionFunc` and `RayClusterCompletionFunc`.

Comment (andrewsykim) [reply kubectl-plugin/pkg/util/completion/completion.go:37]: okay that makes sense to me, thanks for clarifying

**Review by andrewsykim — APPROVED** [review]
LGTM

**Pull request merged**